app/controllers/users_controller.rb

```diff
@@ -17,7 +17,11 @@ def create
   end
 
   def show
-    @user = User.find(session[:user_id])
+    if session[:user_id] == nil
+      render file: "/public/404"
+    else
+      @user = User.find(session[:user_id])
+    end
   end
 
   private
```

Comment:
Even though we are permitting that password confirmation come through our Users Controller, do we need that here?  My gut says yes because we need to confirm that password is good in our model, but I am wondering if that is done prior to any of this information hitting our params?  To clarify this question: Do we need to permit that password_confirmation comes through our params IF our user model is handling the uniqueness the moment the user clicks create user?